include/npcomp/Dialect/Basicpy/IR/BasicpyOps.td

```diff
@@ -230,6 +235,115 @@ def Basicpy_BytesConstantOp : Basicpy_Op<"bytes_constant", [
   let hasFolder = 1;
 }
 
+def Basicpy_SingletonOp : Basicpy_Op<"singleton", [
+    ConstantLike, NoSideEffect]> {
+  let summary = "Constant value for a singleton type";
+  let description = [{
+    Some types only have a single possible value, represented by the
+    SingletonAttr. This op allows creating constants of these types.
+  }];
+  let arguments = (ins);
+  let results = (outs
+    Basicpy_SingletonType:$result
+  );
+  let assemblyFormat = "attr-dict `:` type($result)";
+  let hasFolder = 1;
+}
+
+def Basicpy_StrConstantOp : Basicpy_Op<"str_constant", [
+    ConstantLike, NoSideEffect]> {
+  let summary = "Constant string value";
+  let description = [{
+    A string value of StrType. The value is represented by a StringAttr
+    that is UTF-8 encoded.
+  }];
+  let arguments = (ins
+    StrAttr:$value
+  );
+  let results = (outs
+    Basicpy_StrType:$result
+  );
+  let assemblyFormat = "$value attr-dict";
+  let hasFolder = 1;
+}
+
+//===----------------------------------------------------------------------===//
+// Casting and coercion operations
+//===----------------------------------------------------------------------===//
+
+def Basicpy_AsPredicateValueOp : Basicpy_Op<"as_predicate_value",
```

Comment:
can we just call this `as_i1`? I don't see any clarity advantage to the current name.